### ISSUE TITLE
Interoperability: r/w MetaTensor's array content with numpy api

### ISSUE THREAD

**Issue by wyli** [Project-MONAI/MONAI#4685]
**Is your feature request related to a problem? Please describe.**
Though this might cause inconsistencies between data and metadata in MetaTensor,
it's still convenient to support updating MetaTensor array content with the numpy API, given that numpy is one of the core dependencies of MONAI. This feature request is mainly trying to address the recent interoperability concerns:

- reading the array into a numpy array as a shortcut of `a.detach().cpu().numpy()` (sharing the array memory as much as possible):
```py
a = MetaTensor([2.0])
assert(isinstance(a.array, numpy.ndarray))
```

- writing the array content
```py
from monai.data import MetaTensor
a = MetaTensor([1.0])
a.array = np.asarray([2.0])
```

cc @SachidanandAlle
